core/src/main/java/org/bitcoinj/crypto/DeterministicKey.java

```diff
@@ -307,8 +309,13 @@ public DeterministicKey encrypt(KeyCrypter keyCrypter, AesKey aesKey, @Nullable
         checkState(privKeyBytes != null, "Private key is not available");
         EncryptedData encryptedPrivateKey = keyCrypter.encrypt(privKeyBytes, aesKey);
         DeterministicKey key = new DeterministicKey(childNumberPath, chainCode, keyCrypter, pub, encryptedPrivateKey, newParent);
-        if (newParent == null)
-            key.setCreationTimeSeconds(getCreationTimeSeconds());
+        if (newParent == null) {
+            Optional<Instant> creationTime = getCreationTime();
+            if (creationTime.isPresent())
+                key.setCreationTime(creationTime.get());
+            else
+                key.clearCreationTime();
```

Comment:
Yes, the `ifPresent*` variants are nice!

Reply:
Heck, it would even boil down to:

```
getCreationTime().ifPresentOrElse(
         key::setCreationTime,
         key::clearCreationTime
     );
```

(I similar, can't remember exactly.)

I really like this!